geopyspark/geotrellis/layer.py

```diff
@@ -992,81 +1020,50 @@ def tile_to_layout(self, layout_definition, resample_method=ResampleMethod.NEARE
-            multiples of 2.
+        if isinstance(layout, LayoutDefinition):
+            srdd = self.srdd.tileToLayout(layout, resample_method)
 
-        Args:
-            end_zoom (int): The zoom level where pyramiding should end. Represents
-                the level that is most zoomed out.
-            start_zoom (int, Optional): The zoom level where pyramiding should begin. Represents
-                the level that is most zoomed in. If None, then will use the zoom level from
-                :meth:`~geopyspark.geotrellis.rdd.TiledRasterLayer.zoom_level`.
-            resample_method (str or :class:`~geopyspark.geotrellis.constants.ResampleMethod`, optional):
-                The resample method to use for the reprojection. If none is specified, then
-                ``ResampleMethods.NEAREST_NEIGHBOR`` is used.
+        elif isinstance(layout, Metadata):
+            if self.layer_metadata.crs != layout.crs:
+                raise ValueError("The layout needs to have the same crs as the TiledRasterLayer")
 
-        Returns:
-            :class:`~geopyspark.geotrellis.layer.Pyramid`.
+            srdd = self.srdd.tileToLayout(layout.layout_definition, resample_method)
 
-        Raises:
-            ValueError: If the col and row count is not a power of 2.
-        """
+        elif isinstance(layout, TiledRasterLayer):
+            if self.layer_metadata.crs != layout.layer_metadata.crs:
+                raise ValueError("The layout needs to have the same crs as the TiledRasterLayer")
 
-        num_cols = self.layer_metadata.tile_layout.tileCols
-        num_rows = self.layer_metadata.tile_layout.tileRows
+            metadata = layout.layer_metadata
+            srdd = self.srdd.tileToLayout(metadata.layout_definition, resample_method)
 
-        if (num_cols & (num_cols - 1)) != 0 or (num_rows & (num_rows - 1)) != 0:
-            raise ValueError("Tiles dimensions must powers of 2.  Consider `pyramid_non_power_of_two`.")
+        elif isinstance(layout, (LocalLayout, GlobalLayout)):
+            srdd = self.srdd.tileToLayout(layout, resample_method)
 
-        if start_zoom:
-            start_zoom = start_zoom
-        elif self.zoom_level:
-            start_zoom = self.zoom_level
         else:
-            raise ValueError("No start_zoom given.")
-
-        result = self.srdd.pyramid(start_zoom, end_zoom, ResampleMethod(resample_method).value)
+            raise TypeError("Could not retile from the given layout", layout)
 
-        return Pyramid([TiledRasterLayer(self.pysc, self.layer_type, srdd) for srdd in result])
+        return TiledRasterLayer(self.pysc, self.layer_type, srdd)
 
-    def pyramid_non_power_of_two(self, col_power, row_power, end_zoom, start_zoom=None, resample_method=ResampleMethod.NEAREST_NEIGHBOR):
-        """Creates a ``Pyramid`` instance. This will work even if the tiles' sizes are not
-        multiples of 2.
+    def pyramid(self, resample_method=ResampleMethod.NEAREST_NEIGHBOR):
```

Comment:
I see. Those are good points. Do you think a method should be added to `Pyramid` that lets the user select certain zoom levels?

Something like this:

```python

def select_levels(self, min_zoom=0, max_zoom=None):
    max_zoom = max_zoom or self.max_zoom
```